lib/backend/backend.go

```diff
@@ -244,6 +245,27 @@ func NextPaginationKey(r types.Resource) string {
 	return string(nextKey([]byte(r.GetName())))
 }
 
+// MaskKey masks the given key if any of the given sensitive prefixes are detected.
+// If no sensitive prefixes are given, then defaults will be used.
+// e.g "/tokens/123456789" -> "/tokens/******789"
+func MaskKey(key []byte, sensitivePrefixes ...string) []byte {
+	if len(sensitivePrefixes) == 0 {
+		sensitivePrefixes = sensitiveBackendPrefixes
+	}
+
+	// If a prefix is found in the key parts, mask the subsequent key part (presumably the key name).
+	parts := bytes.Split(key, []byte{Separator})
+	for i := 0; i < len(parts)-1; i++ {
```

Comment:
I'm a bit worried this approach might cause accidental maskings, especially in conjunction with the fact the function now gives the appearance of being available for general use. In the original code the sensitive prefix was compared with the first key only, i.e. in `/a/b/c` only `a` were checked to be sensitive.

With this approach, suppose there was a role named `tokens` stored under the backend key `/roles/tokens/params`, `MaskKey` would then unexpectedly return something like `/roles/tokens/****ms`.